module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,60 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+void printMap(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    for (const auto& arr : bitmap) {
+        for (auto& el : arr) {
+            if (el <= ASCIIno32)
+                std::cout << " ";
+            else
+                std::cout << el;
+        }
+        std::cout << "\n";
+    }
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& grayScale) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(width * height);
+    int count = 0;
+    int tmp = 0;
+
+    for (const auto& row : grayScale) {
+        auto _prev = row.at(0);
+        for (const auto& column : row) {
+            if (_prev == column) {
+                count++;
+                _prev = column;
+                continue;
+            }
+            compressed.push_back(std::make_pair(_prev, count));
+            _prev = column;
+            count = 1;
+        }
+        compressed.push_back(std::make_pair(_prev, count));
+        count = 0;
+    }
```

Comment:
You should rearrange a loop to use push_back just once, also in this case emplace_back is better choice :) 
Another thing is that instead of make_pair you can use just {}